pandas/io/formats/format.py

```diff
@@ -1573,12 +1573,20 @@ def __init__(self, obj, path_or_buf=None, sep=",", na_rep='',
 
     def save(self):
         # create the writer & save
+        if self.encoding is None:
+            if compat.PY2:
+                encoding = 'ascii'
+            else:
+                encoding = 'utf-8'
+        else:
+            encoding = self.encoding
```

Comment:
Not sure this is overkill, but you can do `encoding = self.encoding or ('ascii' if compat.PY2 else 'utf-8')` to condense this whole block of logic.

Reply:
IMO, matter of taste here.  I prefer the block-logic because its easier to read I find.